mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisLanguageDriverAutoConfigurationTest.java

```diff
@@ -244,121 +231,113 @@ void testCustomThymeleafConfigUsingConfigurationProperty() {
-        .isEqualTo(RuntimeConstants.ENCODING_DEFAULT);
-
+        .run(context -> {
+          VelocityLanguageDriver driver = context.getBean(VelocityLanguageDriver.class);
+          @SuppressWarnings("unused")
+          class Param {
+            private Integer id;
+            private Integer version;
+          }
+          Param params = new Param();
+          params.id = 10;
+          params.version = 20;
+          SqlSource sqlSource = driver.createSqlSource(new Configuration(), "#now()", Param.class);
+          BoundSql boundSql = sqlSource.getBoundSql(params);
+          assertThat(boundSql.getSql()).isEqualTo("SELECT CURRENT_TIMESTAMP");
+          VelocityLanguageDriverConfig config = context.getBean(VelocityLanguageDriverConfig.class);
+          @SuppressWarnings("deprecation")
+          String[] userDirective = config.getUserdirective();
+          assertThat(userDirective).hasSize(1).contains(NowDirective.class.getName());
+          assertThat(config.getAdditionalContextAttributes()).hasSize(2);
+          assertThat(config.getAdditionalContextAttributes()).containsEntry("attribute1", "java.lang.String");
+          assertThat(config.getAdditionalContextAttributes()).containsEntry("attribute2", "java.util.HashMap");
+          assertThat(config.getVelocitySettings()).hasSize(3);
+          assertThat(config.getVelocitySettings()).containsEntry(RuntimeConstants.RESOURCE_LOADERS, "class");
+          assertThat(config.getVelocitySettings()).containsEntry(RuntimeConstants.RESOURCE_LOADER + ".class.class",
+              "org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader");
+          assertThat(config.generateCustomDirectivesString()).isEqualTo(NowDirective.class.getName()
+              + ",org.mybatis.scripting.velocity.TrimDirective,org.mybatis.scripting.velocity.WhereDirective,org.mybatis.scripting.velocity.SetDirective,org.mybatis.scripting.velocity.InDirective,org.mybatis.scripting.velocity.RepeatDirective");
+          assertThat(config.getVelocitySettings()).containsEntry(RuntimeConstants.INPUT_ENCODING,
+              RuntimeConstants.ENCODING_DEFAULT);
+        });
   }
 
   @Test
   void testExcludeMybatisLanguageDriverAutoConfiguration() {
-    TestPropertyValues
-        .of("spring.autoconfigure.exclude:org.mybatis.spring.boot.autoconfigure.MybatisLanguageDriverAutoConfiguration")
-        .applyTo(this.context);
-    this.context.register(MyAutoConfiguration.class);
-    this.context.refresh();
-    assertThat(this.context.getBeanNamesForType(LanguageDriver.class)).hasSize(0);
+    new ApplicationContextRunner().withUserConfiguration(MyAutoConfiguration.class)
+        .run(context -> assertThat(context.getBeanNamesForType(LanguageDriver.class)).isEmpty());
   }
 
-  @EnableAutoConfiguration
-  private static class MyAutoConfiguration {
+  @org.springframework.context.annotation.Configuration
+  static class MyAutoConfiguration {
```

Comment:
@eddumelendez 

You can avoid error using `@EnableConfigurationProperties` because theses tests use configuration properties feature.